samples/constitutions/roles/set_role_definition.js

```diff
@@ -0,0 +1,60 @@
+// A simple wrapper for set usage in the KV in the constitution
+class KVSet {
+  #map;
+
+  constructor(map) {
+    this.#map = map;
+  }
+
+  has(key) {
+    return this.#map.has(key);
+  }
+
+  add(key) {
+    this.#map.set(key, new ArrayBuffer(8));
+  }
+
+  delete(key) {
+    this.#map.delete(key);
+  }
+
+  clear() {
+    this.#map.clear();
+  }
+
+  asSetOfStrings() {
+    let set = new Set();
+    this.#map.forEach((_, key) => set.add(ccf.bufToStr(key)));
+    return set;
+  }
+}
+
+actions.set(
+  "set_role_definition",
+  new Action(
+    function (args) {
+      checkType(args.role, "string", "role");
+      checkType(args.actions, "array", "actions");
+      for (const [i, action] of args.actions.entries()) {
+        checkType(action, "string", `actions[${i}]`);
+      }
+    },
+    function (args) {
+      let roleDefinition = new KVSet(
```

Comment:
qq: why is this wrapper needed? is KVSet not already defined in CCF?

Reply:
There is a `kv::Set` internally in C++, but there is a fairly substantial amount of glue between `kv::Map` in C++ and `ccf.kv` in JS, especially post-interpreter re-use (to deal with potential dangling references in context etc).

Even if we essentially duplicated this glue to support both types, there would still be the problem of how to expose it next to `ccf.kv[]`. Open to ideas, but my intention at this point is to look at moving the constitution to TypeScript, because that allows using `typedKvSet` (see #6320), which is quite easy to use but does not require extending a large amount of C++ infra types.